src/index.ts

```diff
@@ -41,12 +42,22 @@ export class WaxJS {
         sign: async (data: any) => {
           return {
             serializedTransaction: data.serializedTransaction,
-            signatures: await this.signing(data.serializedTransaction)
+            signatures: await this.signing(
+              this.signingWindow,
+              data.serializedTransaction
+            )
           };
         }
       };
       // @ts-ignore
       this.api = new Api({ rpc: this.rpc, signatureProvider: signer });
+      const transact = this.api.transact.bind(this.api);
+      const url = this.waxSigningURL + "/cloud-wallet/signing/";
+      this.api.transact = async (...args) => {
```

Comment:
Evertything looks good. The only thing is probably making clear why we do this monkey patch. Maybe add some explanation like:

`// We monkeypatch the transact method to overcome timeouts firing the pop-up which some browsers enforce, such as Safari. By pre-creating the pop-up window we will interact with, we ensure that it is not going to be rejected due to a delayed pop up that would otherwise occur post transaction creation`